src/couch_replicator_httpc.erl

```diff
@@ -169,13 +169,31 @@ process_stream_response(ReqId, Worker, HttpDb, Params, Callback) ->
 % on the ibrowse_req_id format. This just drops all
 % messages for the given ReqId on the floor since we're
 % no longer in the HTTP request.
-clean_mailbox({ibrowse_req_id, ReqId}) ->
+
+clean_mailbox(ReqId) ->
+    clean_mailbox(ReqId, 16).
+
+
+clean_mailbox(_ReqId, N) when N =< 0 ->
```

Comment:
could we match on 0 here and > 0 in the one below? if N was ever below 0 we'd wanna crash ugly